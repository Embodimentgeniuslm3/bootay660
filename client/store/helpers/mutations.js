import cuid from 'cuid';
import each from 'lodash/each';
import Resource from './resource';
import Vue from 'vue';

export default function ($apiUrl) {
  let api = new Resource($apiUrl);

  const fetch = (state, result) => {
    each(result, it => {
      // TODO: Extract api unrelated methods
      api.setCid(it);
      Vue.set(state.items, it._cid, it);
    });
  };

  const reset = (state, result) => {
    state.items = result;
  };

  const add = (state, model) => {
    if (!model._cid) model._cid = cuid();
    Vue.set(state.items, model._cid, model);
  };

  const save = (state, model) => {
    Vue.set(state.items, model._cid, model);
  };

  const remove = (state, result) => {
    result.forEach(it => Vue.delete(state.items, it._cid));
  };

  const update = (state, model) => {
    Vue.set(state.items, model._cid, model);
  };

  const setBaseUrl = (state, url) => {
    state.$baseUrl = url;
  };

  return {
    api,
    fetch,
    reset,
    add,
    save,
    remove,
    update,
    setBaseUrl
  };
}
