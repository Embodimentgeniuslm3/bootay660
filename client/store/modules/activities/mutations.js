import generateMutations from '../../helpers/mutations';

const { fetch, reset, add, save, remove, update, setBaseUrl } = generateMutations();

const reorder = (state, { activity, position }) => {
  activity.position = position;
};

export { fetch, reset, add, save, remove, reorder, update, setBaseUrl };
