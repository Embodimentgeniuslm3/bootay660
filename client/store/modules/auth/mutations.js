export const setAuth = (state, { user, authStrategy }) => {
  state.user = user;
  state.strategy = authStrategy;
};

export const resetAuth = state => {
  state.user = null;
  state.strategy = null;
};
