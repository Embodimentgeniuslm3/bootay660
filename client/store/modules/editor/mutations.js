const setIsPublishedPreview = (state, isPublishedPreview) => {
  state.isPublishedPreview = isPublishedPreview;
};

export { setIsPublishedPreview };
