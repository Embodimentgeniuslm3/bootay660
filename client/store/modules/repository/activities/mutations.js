import {
  add, fetch, remove, reset, save, setEndpoint
} from '@/store/helpers/mutations';

const reorder = (state, { activity, position }) => {
  state.items[activity._cid].position = position;
};

export { add, fetch, remove, reorder, reset, save, setEndpoint };
