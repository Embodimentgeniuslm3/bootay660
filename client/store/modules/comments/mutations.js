import { fetch, remove, reset, save, setEndpoint } from '../../helpers/mutations';
import find from 'lodash/find';
import pick from 'lodash/pick';
import Vue from 'vue';

const setRepository = (state, repositoryId) => {
  state.repositoryId = repositoryId;
  state.activitiesFetched = {};
};

const commentsFetched = (state, activityId) => {
  Vue.set(state.activitiesFetched, activityId, true);
};

const sseUpdate = (state, comment) => {
  const existing = find(state.items, { id: comment.id });
  if (!existing) return;
  const data = pick(comment, ['content', 'createdAt', 'updatedAt', 'deletedAt']);
  Vue.set(state.items, existing._cid, { ...existing, ...data });
};

export {
  commentsFetched,
  fetch,
  remove,
  reset,
  save,
  setRepository,
  setEndpoint,
  sseUpdate
};
