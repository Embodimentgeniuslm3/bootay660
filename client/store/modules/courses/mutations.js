import { add, fetch, remove, reset, setEndpoint } from '../../helpers/mutations';
import Vue from 'vue';

const PAGINATION_DEFAULTS = { offset: 0, limit: 21 };

const save = (state, course) => {
  const search = state.search && state.search.toLowerCase();
  const name = course.name.toLowerCase();
  if (search && !name.includes(search)) return;
  Vue.set(state.items, course._cid, course);
};

const fetchPinned = (state, courses) => {
  state.pinned = courses;
};

const togglePinned = state => {
  state.showPinned = !state.showPinned;
};

const setSearch = (state, query = '') => {
  state.$internals.pagination = PAGINATION_DEFAULTS;
  state.search = query;
};

const setOrder = (state, order) => {
  state.$internals.sort = { ...state.$internals.sort, ...order };
};

const setPagination = (state, changes) => {
  const $internals = state.$internals;
  $internals.pagination = { ...$internals.pagination, ...changes };
};

const resetPagination = state => {
  state.$internals.pagination = PAGINATION_DEFAULTS;
};

const allCoursesFetched = (state, allFetched) => {
  state.$internals.allCoursesFetched = allFetched;
};

export {
  add,
  allCoursesFetched,
  fetch,
  fetchPinned,
  remove,
  reset,
  resetPagination,
  save,
  setEndpoint,
  setPagination,
  setOrder,
  setSearch,
  togglePinned
};
