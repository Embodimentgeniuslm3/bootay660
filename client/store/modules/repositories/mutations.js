import { add, fetch, remove, reset, setEndpoint } from '../../helpers/mutations';
import find from 'lodash/find';
import findIndex from 'lodash/findIndex';
import Vue from 'vue';

const PAGINATION_DEFAULTS = { offset: 0, limit: 21 };

const save = (state, repository) => {
  const search = state.search && state.search.toLowerCase();
  const name = repository.name.toLowerCase();
  if (search && !name.includes(search)) return;
  Vue.set(state.items, repository._cid, repository);
};

const togglePinned = state => {
  resetPagination(state);
  state.showPinned = !state.showPinned;
};

const setSearch = (state, query = '') => {
  resetPagination(state);
  state.search = query;
};

const setOrder = (state, order) => {
  resetPagination(state);
  state.$internals.sort = { ...state.$internals.sort, ...order };
};

const setPagination = (state, changes) => {
  const $internals = state.$internals;
  $internals.pagination = { ...$internals.pagination, ...changes };
};

const resetPagination = state => {
  state.$internals.pagination = { ...PAGINATION_DEFAULTS };
};

const resetFilters = state => {
  state.search = '';
  state.showPinned = false;
};

const allRepositoriesFetched = (state, allFetched) => {
  state.$internals.allRepositoriesFetched = allFetched;
};

const fetchTags = (state, tags) => {
  state.tags = tags;
};

const saveTags = (state, { tags, data }) => {
  const repo = find(state.items, { id: data.repositoryId });
  state.tags.push(data);
  repo.tags.push(tags.tag);
};

const removeTag = (state, tag) => {
  const repo = find(state.items, { id: tag.repositoryId });
  const tagIndex = findIndex(repo.tags, { id: tag.tagId });
  repo.tags.splice(tagIndex, 1);
};

export {
  add,
  allRepositoriesFetched,
  fetch,
  fetchTags,
  remove,
  removeTag,
  reset,
  resetPagination,
  resetFilters,
  save,
  saveTags,
  setEndpoint,
  setPagination,
  setOrder,
  setSearch,
  togglePinned
};
