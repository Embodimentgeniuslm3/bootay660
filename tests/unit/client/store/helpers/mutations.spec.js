import { add, fetch, remove, reset, save, setEndpoint } from '@/client/store/helpers/mutations.js';
import { mockCommentModel, mockCommentState, mockItems } from '../__mocks__/mockCommentData';

jest.mock('cuid');
// eslint-disable-next-line no-unused-vars
const cuid = require('cuid').mockImplementation(() => 'newCuid');

describe('Helpers - mutation test', () => {
  it('should test fetch mutation sets the vue instance state', () => {
    const state = mockCommentState();
    const items = mockItems();
    fetch(state, items);
    expect(state).toMatchSnapshot();
  });

  it('should reset mutation resets the comments in the vue instance', () => {
    const state = mockCommentState();
    const items = mockItems();
    reset(state, items);
    expect(state).toMatchSnapshot();
    const newState = mockCommentState();
    reset(newState);
    expect(newState.items).toEqual({});
  });

  it('should add a item to the state', () => {
    let state = mockCommentState();
    const model = mockCommentModel();
    add(state, model);
    expect(state).toMatchSnapshot();
    delete model._cid;

    state = mockCommentState();
    add(state, model);
    expect(state).toMatchSnapshot();
  });

  it('should remove a item from the state', () => {
    const state = mockCommentState();
    const model1 = mockCommentModel();
    const model2 = mockCommentModel('hereisAfakeCuild');
    add(state, model1);
    add(state, model2);
    remove(state, [model1]);
    expect(state.items).toHaveProperty('hereisAfakeCuild');
  });

  it('should save item to the state', () => {
    const state = mockCommentState();
    const model = mockCommentModel();
    save(state, model);
    expect(state).toMatchSnapshot();
  });

  it('test setting a end point for the apiUrl the state', () => {
    const state = mockCommentState();
    const url = '/some/ulr';
    setEndpoint(state, url);
    expect(state.$apiUrl).toEqual(url);
  });
});
