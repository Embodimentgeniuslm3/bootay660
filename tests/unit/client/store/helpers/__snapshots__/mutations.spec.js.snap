// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers - mutation test should add a item to the state 1`] = `
Object {
  "$apiUrl": "/repositories/1/activities",
  "$internals": Object {},
  "items": Object {
    "ck6i4v82h00063h5wvhr7u0uc": Object {
      "_cid": "ck6i4v82h00063h5wvhr7u0uc",
      "_synced": false,
      "_version": 1581440542361,
      "activityId": 63,
      "author": Object {
        "createdAt": "2020-02-03T14:44:03.815Z",
        "deletedAt": null,
        "email": "admin@extensionengine.com",
        "firstName": null,
        "fullName": null,
        "id": 1,
        "imgUrl": "https://s.gravatar.com/avatar/b29ca01752b9c16029d61c1c8c8094cc?size=130&default=identicon",
        "label": "admin@extensionengine.com",
        "lastName": null,
        "role": "ADMIN",
        "updatedAt": "2020-02-03T14:44:03.815Z",
      },
      "content": "Here is the comment text",
      "createdAt": 1581440542361,
      "updatedAt": 1581440542361,
    },
  },
}
`;

exports[`Helpers - mutation test should add a item to the state 2`] = `
Object {
  "$apiUrl": "/repositories/1/activities",
  "$internals": Object {},
  "items": Object {
    "newCuid": Object {
      "_cid": "newCuid",
      "_synced": false,
      "_version": 1581440542361,
      "activityId": 63,
      "author": Object {
        "createdAt": "2020-02-03T14:44:03.815Z",
        "deletedAt": null,
        "email": "admin@extensionengine.com",
        "firstName": null,
        "fullName": null,
        "id": 1,
        "imgUrl": "https://s.gravatar.com/avatar/b29ca01752b9c16029d61c1c8c8094cc?size=130&default=identicon",
        "label": "admin@extensionengine.com",
        "lastName": null,
        "role": "ADMIN",
        "updatedAt": "2020-02-03T14:44:03.815Z",
      },
      "content": "Here is the comment text",
      "createdAt": 1581440542361,
      "updatedAt": 1581440542361,
    },
  },
}
`;

exports[`Helpers - mutation test should reset mutation resets the comments in the vue instance 1`] = `
Object {
  "$apiUrl": "/repositories/1/activities",
  "$internals": Object {},
  "items": Object {
    "ck6i0fsmg00003h5wqtirijkm": Object {
      "_cid": "ck6i0fsmg00003h5wqtirijkm",
      "createdAt": "2020-02-05T17:44:29.055Z",
      "data": Object {
        "name": "stolar",
      },
      "deletedAt": null,
      "detached": false,
      "id": 63,
      "parentId": null,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "DEFAULT_SCHEMA/COMPETENCY",
      "uid": "570149c9-52e1-46b9-8946-a55bdd636c9b",
      "updatedAt": "2020-02-05T17:44:29.055Z",
    },
    "ck6i0fsmg00013h5w7hmh3r3a": Object {
      "_cid": "ck6i0fsmg00013h5w7hmh3r3a",
      "createdAt": "2020-02-05T21:02:07.902Z",
      "data": Object {},
      "deletedAt": null,
      "detached": false,
      "id": 69,
      "parentId": 63,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "INTRO",
      "uid": "9de9f40f-1a40-46fd-9ac6-3773cfea1e82",
      "updatedAt": "2020-02-05T21:02:07.902Z",
    },
    "ck6i0fsmg00023h5wqbtdj1e1": Object {
      "_cid": "ck6i0fsmg00023h5wqbtdj1e1",
      "createdAt": "2020-02-06T13:27:21.611Z",
      "data": Object {
        "name": "prvi",
      },
      "deletedAt": null,
      "detached": false,
      "id": 80,
      "parentId": 63,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "DEFAULT_SCHEMA/OBJECTIVE",
      "uid": "25f656bb-990f-4ec1-978b-c5ff868ad11c",
      "updatedAt": "2020-02-06T13:27:21.611Z",
    },
  },
}
`;

exports[`Helpers - mutation test should save item to the state 1`] = `
Object {
  "$apiUrl": "/repositories/1/activities",
  "$internals": Object {},
  "items": Object {
    "ck6i4v82h00063h5wvhr7u0uc": Object {
      "_cid": "ck6i4v82h00063h5wvhr7u0uc",
      "_synced": false,
      "_version": 1581440542361,
      "activityId": 63,
      "author": Object {
        "createdAt": "2020-02-03T14:44:03.815Z",
        "deletedAt": null,
        "email": "admin@extensionengine.com",
        "firstName": null,
        "fullName": null,
        "id": 1,
        "imgUrl": "https://s.gravatar.com/avatar/b29ca01752b9c16029d61c1c8c8094cc?size=130&default=identicon",
        "label": "admin@extensionengine.com",
        "lastName": null,
        "role": "ADMIN",
        "updatedAt": "2020-02-03T14:44:03.815Z",
      },
      "content": "Here is the comment text",
      "createdAt": 1581440542361,
      "updatedAt": 1581440542361,
    },
  },
}
`;

exports[`Helpers - mutation test should test fetch mutation sets the vue instance state 1`] = `
Object {
  "$apiUrl": "/repositories/1/activities",
  "$internals": Object {},
  "items": Object {
    "ck6i0fsmg00003h5wqtirijkm": Object {
      "_cid": "ck6i0fsmg00003h5wqtirijkm",
      "createdAt": "2020-02-05T17:44:29.055Z",
      "data": Object {
        "name": "stolar",
      },
      "deletedAt": null,
      "detached": false,
      "id": 63,
      "parentId": null,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "DEFAULT_SCHEMA/COMPETENCY",
      "uid": "570149c9-52e1-46b9-8946-a55bdd636c9b",
      "updatedAt": "2020-02-05T17:44:29.055Z",
    },
    "ck6i0fsmg00013h5w7hmh3r3a": Object {
      "_cid": "ck6i0fsmg00013h5w7hmh3r3a",
      "createdAt": "2020-02-05T21:02:07.902Z",
      "data": Object {},
      "deletedAt": null,
      "detached": false,
      "id": 69,
      "parentId": 63,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "INTRO",
      "uid": "9de9f40f-1a40-46fd-9ac6-3773cfea1e82",
      "updatedAt": "2020-02-05T21:02:07.902Z",
    },
    "ck6i0fsmg00023h5wqbtdj1e1": Object {
      "_cid": "ck6i0fsmg00023h5wqbtdj1e1",
      "createdAt": "2020-02-06T13:27:21.611Z",
      "data": Object {
        "name": "prvi",
      },
      "deletedAt": null,
      "detached": false,
      "id": 80,
      "parentId": 63,
      "position": 1,
      "publishedAt": null,
      "refs": Object {},
      "repositoryId": 1,
      "type": "DEFAULT_SCHEMA/OBJECTIVE",
      "uid": "25f656bb-990f-4ec1-978b-c5ff868ad11c",
      "updatedAt": "2020-02-06T13:27:21.611Z",
    },
  },
}
`;
